### PR TITLE
[2.1] Prevent crash for invalid reference

### PR DESCRIPTION
**Pull request by guilhermefelipecgs**
Fix for 2.1 branch.

For 3.1 I'm fixing here #18716.

### CODE REVIEW

File: platform/osx/os_osx.mm [hunk 1]
```diff
@@ -1257,9 +1257,9 @@ static void displays_arrangement_changed(CGDirectDisplayID display_id, CGDisplay
 			uint32_t color = image.get_pixel(column_index, row_index).to_ARGB32();
 
 			uint8_t alpha = (color >> 24) & 0xFF;
-			pixels[i * 4 + 0] = ((color >> 16)  & 0xFF) * alpha / 255;
+			pixels[i * 4 + 0] = ((color >> 16) & 0xFF) * alpha / 255;
 			pixels[i * 4 + 1] = ((color >> 8) & 0xFF) * alpha / 255;
-			pixels[i * 4 + 2] = ((color) & 0xFF) * alpha / 255;
+			pixels[i * 4 + 2] = ((color)&0xFF) * alpha / 255;
```
Comment (akien-mga) [inline platform/osx/os_osx.mm]: Missing spaces around operator.

File: platform/windows/os_windows.cpp [hunk 2]
```diff
@@ -2003,7 +2003,7 @@ void OS_Windows::set_custom_mouse_cursor(const RES &p_cursor, CursorShape p_shap
 
 			texture_size.width = atlas_texture->get_region().size.x;
 			texture_size.height = atlas_texture->get_region().size.y;
-		} else {
+		} else if(texture.is_valid()) {
```
Comment (akien-mga) [inline platform/windows/os_windows.cpp]: There should be a space after `if`.

File: platform/x11/os_x11.cpp [hunk 3]
```diff
@@ -2177,7 +2177,7 @@ void OS_X11::set_custom_mouse_cursor(const RES &p_cursor, CursorShape p_shape, c
 
 			texture_size.width = atlas_texture->get_region().size.x;
 			texture_size.height = atlas_texture->get_region().size.y;
-		} else {
+		} else if(texture.is_valid()) {
```
Comment (akien-mga) [inline platform/x11/os_x11.cpp]: Same here.

File: platform/osx/os_osx.mm (hunk 1 above)
Comment (guilhermefelipecgs) [reply platform/osx/os_osx.mm]: Fixed.

**Pull request merged**